runtime/src/lib.rs

```diff
@@ -20,6 +20,10 @@ pub use slices::{
 mod result;
 pub use result::DiplomatResult;
 
+/// A type to represent Option<T> over FFI. Prefer using Option<T> with boxes and reference
+/// types, and in return types, use this in structs/fn args.
```

Comment:
It would be nice if args would get auto-translated the same way that return values do. Then users would only have to worry about the `Option`/`DiplomatOption` reference thing in structs.

That said, we *could* also auto-translate struct fields, and not make users ever write `DiplomatOption`, the same way that `DiplomatResult` is under the hood.

Reply:
I no args are already auto translated in this PR? Option is marked as non FFI safe which means it's allowed in args but not fields.

Reply:
Oh yeah I can fix that.

I'm definitely going to write precise docs in the book after this.